<a href="https://colab.research.google.com/github/duellal/DS-Unit-1-Sprint-2-Statistics/blob/master/Duell_LS_DS_124_Introduction_to_Bayesian_Inference_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




## Assignment - Code it up!

We used pure math to apply Bayes Theorem to drug tests. Now write Python code to reproduce the results! This is purposefully open ended - you'll have to think about how you should represent probabilities and events. You can and should look things up.

Specific goals/targets:

### 1) Write a function 

`def prob_drunk_given_positive(prob_drunk_prior, false_positive_rate, true_positive_rate):` 

You should only truly need these three values in order to apply Bayes Theorem. In this example, imagine that individuals are taking a breathalyzer test with an 8% false positive rate, a 100% true positive rate, and that our prior belief about drunk driving in the population is 1/1000. 
 - What is the probability that a person is drunk after one positive breathalyzer test?
 - What is the probability that a person is drunk after two positive breathalyzer tests?
 - How many positive breathalyzer tests are needed in order to have a probability that's greater than 95% that a person is drunk beyond the legal limit?


In [ ]:
def prob_drunk_positive(prob_drunk_prior, false_pos_rate = .08, true_pos_rate = 1):
  population = (1-prob_drunk_prior)
  numerator = (true_pos_rate * prob_drunk_prior)
  denom = ((true_pos_rate * prob_drunk_prior) + (false_pos_rate * population))
  return numerator/denom

In [ ]:
prob_drunk_positive(.001)

0.012357884330202669

After one positive breathalyzer test, the probability that the person is drunk is 1.24%.

In [ ]:
prob_drunk_positive(0.012357884330202669)

0.13525210993291495

After two positive breathalyzer tests, the probability that the person is drunk is 13.5%.

In [ ]:
prob_drunk_positive(0.13525210993291495)

0.6615996951348605

In [ ]:
prob_drunk_positive(0.6615996951348605)

0.9606895076105054

There will need to be 4 breathalyzer tests in order to state that the person is drunk beyond the legal limit with at least 95% confidence.

### 2) In your own words, summarize the difference between Bayesian and Frequentist statistics

If you're unsure where to start, check out [this blog post of Bayes theorem with Python](https://dataconomy.com/2015/02/introduction-to-bayes-theorem-with-python/).

Frequentist statistics is where you base your probable outcomes on a specific non-changing sample. 

Bayesian statistics is where you base your probable outcomes on previous outcomes or previous beliefs. You can "update" your belief based on the new evidence produced.

### 3) Use the following Template to help come up with ideas for your Build Sprint Project: 



---

## Idea 1:

### You
  1. What do you care about? 

  The Education System

  2. What do you know about? 

  The pay and time it takes being a teacher.

  3. What decisions do you face? 
  
  Is it worth it being a teacher?




### Seven templates

In order to better understand the prompts here, please review the [Priceonomics Content Marketing Templates](https://priceonomics.com/introducing-priceonomics-content-marketing/)

Can you apply the templates to your topics?

**Geographic Variation:** Where in the US do teachers get paid more? Less?

**Trend related to the news:** Teachers' salaries are low compared to other similar jobs in America.

**Who does that?:** What types of schools pay more?

**Answering a question people care about:** Are teachers' salaries enough to care for your family? (est. family size is 1 - 4)

**Valuable to businesses:** What type of education are parents looking for?

**What's the most popular?:** What is the highest teacher salary?

**Cost/Money rankings:** What degree of education do teachers who make the most get?

### Misconceptions

What misconceptions do people have about your topic?

*   Teachers have it easy
*   The amount of salary a teacher makes is enough to not be in poverty.
*   Teachers only work during the school day.
*   Teachers get the summer off.

### Examples

1.  What data storytelling example inspires you?

    Visual representation using graphs with hard data.

2.  Could you do a new hypothesis, for the same question?

    Teachers in America are undervalued and underpaid.

3.  Could you do a new question, for the same topic?

    Have teacher salaries changed over time in relation to the economy.

4.  Could you do a new topic, with the same "style"?

###Data

Where could you search for data about your topic?

*   Census data
*   master dataset site
*   government dataset
*   State government dataset



---

## Idea 2:

### You
*   What do you care about?

    Eradicating Diseases
    
*   What do you know about?

    Research is funded by private and public entities.

*   What decisions do you face?

    How many disease I look at. 

    Diseases funded the same amount.

    Diseases funded for the same amount of time.
  

Does how much money is put into the research and vaccinations for known diseases correlate to the elimination or vaccination for those diseases?

### Seven templates

In order to better understand the prompts here, please review the [Priceonomics Content Marketing Templates](https://priceonomics.com/introducing-priceonomics-content-marketing/)

Can you apply the templates to your topics?

**Geographic Variation:** Does one part of America recieve more funding? Do they have more diseases in that region?

**Trend related to the news:** Coronavirus vaccines.

**Who does that?:** Based on funding, when a vaccine is released, how many people are getting a vaccine for that said disease?

How many people are able to get the vaccine or medicine for the disease?

**Answering a question people care about:** Does the amount of money inputted into disease research make a difference in the outcome of eradicating or getting a vaccine for the said disease?

**Valuable to businesses:** What types of diseases does the average person care about?

Will funding a disease create more customers for the said business?

**What's the most popular?:** What disease to date has had the most funding?

**Cost/Money rankings:** What types of vaccines cost the most money to make?



### Misconceptions

What misconceptions do people have about your topic?



*   Funding automatically lets companies start creating a vaccine for a disease.
*   Less funding will make the process longer.
*   Funding doesn't matter, companies will still try to create a solution for the disease.
*   A lot of the companies that are funded pocket the money instead of using it for research.



### Examples

What data storytelling example inspires you?

Could you do a new hypothesis, for the same question?

Could you do a new question, for the same topic?

Could you do a new topic, with the same "style"?

###Data

Where could you search for data about your topic?

*  specific research websites for a disease.
*  government datasets
*  businesses who have donated
*  state datasets

## Resources

- [Worked example of Bayes rule calculation](https://en.wikipedia.org/wiki/Bayes'_theorem#Examples) (helpful as it fully breaks out the denominator)
- [Source code for mvsdist in scipy](https://github.com/scipy/scipy/blob/90534919e139d2a81c24bf08341734ff41a3db12/scipy/stats/morestats.py#L139)

## Stretch Goals:

- Go back and study the content from Modules 1 & 2 & 3 to make sure that you're really comfortable with them.
- Apply a Bayesian technique to a problem you previously worked (in an assignment or project work) on from a frequentist (standard) perspective
- Check out [PyMC3](https://docs.pymc.io/) (note this goes beyond hypothesis tests into modeling) - read the guides and work through some examples
- Take PyMC3 further - see if you can build something with it!